### PR TITLE
roachpb,ui: upgrade protobufjs for reserved enum field support

### PR DESCRIPTION
**Pull request by benesch**
We noticed in #26935 that protobufjs pre-6.8 choked on reserved enum fields. Upgrade it so that we can mark roachpb.ValueType::NULL as reserved. Unfortunately protobufjs 6.8 decided to change the name of the generated interface type (IFoo instead of Foo$Properties) and so the resulting diff is massive. C'est la vie.

@vilterp and @couchand for the first commit. @mberhault for the second.

### CODE REVIEW

File: pkg/ui/src/protobufInit.ts [hunk 1]
```diff
@@ -1,5 +1,3 @@
-/// <reference path="../node_modules/protobufjs/stub-node.d.ts" />
```
Comment (benesch) [inline pkg/ui/src/protobufInit.ts]: Not entirely sure why this needed to go, but `make lint` was complaining about it.

**Review by vilterp — APPROVED** [review]
LGTM on the first commit. Tested locally; everything seems to still work 😅

**Comment by mberhault** [comment]
LGTM. Thanks!

**Comment by danhhz** [comment]
Review status: :shipit: complete! 0 of 0 LGTMs obtained

---

*[pkg/roachpb/data.proto, line 50 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/27084#-LGQhao10_LMZ5lpSpq1:-LGQhao10_LMZ5lpSpq2:b-uljzqi) ([raw file](https://github.com/cockroachdb/cockroach/blob/6e5b3d067fc17a5400ff95a460529c3203b56822/pkg/roachpb/data.proto#L50)):*
> ```protobuf
>   // variant that self-delimits length.
>   UNKNOWN = 0;
>   reserved 7;
> ```

oh geez. when I mentioned this as an option, I really didn't mean for us to go down a rabbit hole. :flushed: 

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/27084)*

**Comment by benesch** [comment]
> oh geez. when I mentioned this as an option, I really didn't mean for us to go down a rabbit hole. 😳

Never hurts to keep things up to date! 😀

File: pkg/ui/src/protobufInit.ts (hunk 1 above)
Comment (couchand) [reply pkg/ui/src/protobufInit.ts]: Looks like they cleaned some stuff up: https://github.com/dcodeIO/protobuf.js/commit/812b38ddabb35e154f9ff94f32ad8ce2a70310f1

**Comment by couchand** [comment]
I got a couple odd errors while switching over to this, but nothing too bad:

- I had `yarn-vendor` still lingering that I had to delete (I guess you'll need to rebase?)
- It didn't rebuild the protos automatically, I had to `rm src/js/protos.*`
- I got a "Cannot find module 'espree'", but `make` again after that figured itself out.

After that everything seems to work fine.  LGTM

**Comment by benesch** [comment]
> * I had yarn-vendor still lingering that I had to delete (I guess you'll need to rebase?)
Yep, hadn't rebased.

> It didn't rebuild the protos automatically, I had to rm src/js/protos.*

Yeah, this is https://github.com/cockroachdb/cockroach/issues/26363.

> I got a "Cannot find module 'espree'", but make again after that figured itself out.

And this is protobufjs stupidity. I've tried very hard to track this down and it still seems to happen occasionally. Sigh.

**Comment by benesch** [comment]
bors r=vilterp,mberhault,couchand

**Pull request merged**